Sdk/Track/Swrve.m

```diff
@@ -2295,6 +2295,99 @@ - (void) generateShortDeviceId {
     }
 }
 
+-(void)filterLocationCampaigns:(PlotFilterNotifications*)filterNotifications {
+
+    BOOL targetLocationCampaigns = NO;
+    for (UILocalNotification* localNotification in filterNotifications.uiNotifications) {
+        NSString* plotPayload = [localNotification.userInfo objectForKey:PlotNotificationActionKey];
+        if(!plotPayload) {
+            DebugLog(@"Error in filterLocationCampaigns. No payload action.", nil);
+            continue;
+        }
+        NSError *payloadError = nil;
+        NSData *objectData = [plotPayload dataUsingEncoding:NSUTF8StringEncoding];
+        NSMutableDictionary *jsonPayLoad = [NSJSONSerialization JSONObjectWithData:objectData options:NSJSONReadingMutableContainers error:&payloadError];
+        if(payloadError) {
+            DebugLog(@"Error in filterLocationCampaigns. Problem parsing payload action: %@", plotPayload);
+            continue;
+        }
+        NSString* geofenceId = [jsonPayLoad objectForKey:@"geofenceId"];
```

Comment:
You can save an Array of campaigns that have the payload you want, and pass only that to the targetLocationsCampaigns method (to avoid code duplication)